addons/l10n_in_edi/models/account_edi_format.py

```diff
@@ -0,0 +1,442 @@
+            ....
+        }
+        """
+        invoice_line_details = {}
+        for index, line in enumerate(move.invoice_line_ids.filtered(lambda line: not line.display_type), start=1):
+            line_details = invoice_line_tax_details.get(line, {})
+            tax_details = self._get_l10n_in_tax_details_by_line_code(line_details.get("tax_details", {}))
+            invoice_line_details[line] = {
+                "SlNo": str(index),
+                "PrdDesc": line.name.replace("\n", ""),
+                "IsServc": line.product_id.type == "service" and "Y" or "N",
+                "HsnCd": line.product_id.l10n_in_hsn_code,
+                "Qty": self._l10n_in_round_value(line.quantity or 0.0, 3),
+                "Unit": line.product_uom_id.l10n_in_code and line.product_uom_id.l10n_in_code.split("-")[0] or "OTH",
+                # Unit price in company currency and tax excluded so its different then price_unit
+                "UnitPrice": self._l10n_in_round_value(
+                    ((line.balance / (1 - (line.discount / 100))) / line.quantity) * sign, 3),
+                # total amount is before discount
+                "TotAmt": self._l10n_in_round_value((line.balance / (1 - (line.discount / 100))) * sign, 2),
+                "Discount": self._l10n_in_round_value(
+                    ((line.balance / (1 - (line.discount / 100))) - line.balance) * sign, 2),
+                "AssAmt": self._l10n_in_round_value(line.balance * sign, 2),
+                "GstRt": self._l10n_in_round_value(tax_details.get("igst_rate", 0.00) or (
+                    tax_details.get("cgst_rate", 0.00) + tax_details.get("sgst_rate", 0.00)), 3),
+                "IgstAmt": self._l10n_in_round_value(tax_details.get("igst_amount", 0.00) * sign, 2),
+                "CgstAmt": self._l10n_in_round_value(tax_details.get("cgst_amount", 0.00) * sign, 2),
+                "SgstAmt": self._l10n_in_round_value(tax_details.get("sgst_amount", 0.00) * sign, 2),
+                "CesRt": self._l10n_in_round_value(tax_details.get("cess_rate", 0.00), 3),
+                "CesAmt": self._l10n_in_round_value(tax_details.get("cess_amount", 0.00) * sign, 2),
+                "CesNonAdvlAmt": self._l10n_in_round_value(tax_details.get("cess_non_advol_amount", 0.00) * sign, 2),
+                "StateCesRt": self._l10n_in_round_value(tax_details.get("state_cess_rate_amount", 0.00), 3),
+                "StateCesAmt": self._l10n_in_round_value(tax_details.get("state_cess_amount", 0.00) * sign, 2),
+                "StateCesNonAdvlAmt": self._l10n_in_round_value(
+                    tax_details.get("state_cess_non_advol_amount", 0.00) * sign, 2),
+                "OthChrg": self._l10n_in_round_value(tax_details.get("other_amount", 0.00) * sign, 2),
+                "TotItemVal": self._l10n_in_round_value((line.balance + line_details.get("tax_amount", 0.00)) * sign, 2),
+            }
+        return invoice_line_details
+
+    def _l10n_in_edi_generate_json(self, move):
```

Comment:
plz indicate this is about invoice.
- Rename `move` to `invoice`
- Rename the method to `_l10n_in_edi_generate_invoice_json`